src/main/webapp/webapptests/spec/SpecConfirmation.js

```diff
@@ -60,7 +61,13 @@ describe("Confirmation", function() {
         });
 
         it("should have the inner HTML of the h3 tag equal to to the email of the tutor", function() {
-            expect(actual.childNodes[0].childNodes[0].innerHTML).toEqual("Tutoring Session with tester@gmail.com");
+            var tutor = {email: "tester@gmail.com"};
+            spyOn(window, "fetch").and.returnValues(Promise.resolve({json: () => Promise.resolve(user)}), Promise.resolve({json: () => Promise.resolve(tutor)}));
```

Comment:
How does this Promise.resolve() work? I hadn't seen it before

Reply:
It basically returns a new resolved promise object. You can get the value inside the promise using .then(). This might explain it a bit better: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/resolve

Reply:
thanks!